llm-translator.html

```diff
@@ -0,0 +1,276 @@
+        <button id="leave" onclick="leaveRoom()">Leave Room</button>
+      </div>
+      <div id="controls-right">
+        <span>Language:</span>
+        <select id="language" onChange="changeLanguage()">
+          <option value="english" selected>English</option>
+        </select>
+      </div>
+    </div>
+    <div id="videos"></div>
+    <div id="audios"></div>
+    <script>
+      const urlParams = new URLSearchParams(window.location.search);
+
+      const languageSelect = document.getElementById("language");
+      const loadButton = document.querySelector("button#load-devices");
+      const camSelect = document.querySelector("select#cams");
+      const micSelect = document.querySelector("select#mics");
+      const host = !!urlParams.get("host");
+
+      function cap(string) {
+        return string.charAt(0).toUpperCase() + string.slice(1);
+      }
+
+      function isTranslator(p) {
+        return p.user_name.match(/^tb\-.*/);
+      }
+
+      function translatorLanguage(p) {
+        return p.user_name.match(/^tb\-(.*)/)[1];
+      }
+
+      function showSubtitle(paxId, message) {
+        const subtitleDiv = document.getElementById("subtitles-" + paxId);
+        const p = document.createElement("p");
+        p.innerHTML = message;
+        subtitleDiv.appendChild(p);
+        setTimeout(() => {
+          p.remove();
+        }, 7000);
+      }
+
+      async function main() {
+        if (host) {
+          loadButton.style.display = "inline";
+          camSelect.style.display = "inline";
+          micSelect.style.display = "inline";
+        }
+      }
+
+      function startTrack(evt) {
+        if (evt.track.kind === "audio" && !evt.participant.local) {
+          let audiosDiv = document.getElementById("audios");
+          let audioEl = document.createElement("audio");
+          audiosDiv.appendChild(audioEl);
+          audioEl.style.width = "100%";
+          audioEl.srcObject = new MediaStream([evt.track]);
+          audioEl.play();
+        } else if (evt.track.kind === "video") {
+          let pid = evt.participant.session_id;
+          let videosDiv = document.getElementById("videos");
+          let paxDiv = document.createElement("div");
+          paxDiv.id = "pax-" + pid;
+          let videoEl = document.createElement("video");
+          let subtitleDiv = document.createElement("div");
+          subtitleDiv.classList.add("subtitles");
+          subtitleDiv.id = "subtitles-" + pid;
+          paxDiv.appendChild(videoEl);
+          paxDiv.appendChild(subtitleDiv);
+          videosDiv.appendChild(paxDiv);
+          videoEl.srcObject = new MediaStream([evt.track]);
+          videoEl.play();
+        }
+      }
+
+      function stopTrack(evt) {
+        let vids = document.getElementsByTagName("video");
+        for (let vid of vids) {
+          if (
+            vid.srcObject &&
+            vid.srcObject.getVideoTracks()[0] === evt.track
+          ) {
+            vid.remove();
```

Comment:
Should remember to set `srcObject` to `null` for each media element to clean up web media players properly. Otherwise they will leak.